src/quic/preferredaddress.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include <env.h>
+#include <node_internals.h>
+#include <ngtcp2/ngtcp2.h>
+#include <string>
+#include <v8.h>
+
+namespace node {
+namespace quic {
+
+// PreferredAddress is a helper class used only when a client Session receives
+// an advertised preferred address from a server. The helper provides
+// information about the server advertised preferred address and allows
+// the preferred address to be selected.
```

Comment:
Understanding how this utility is going to be used is helpful to reviewing it. There's really only one place in the quic impl where this will be used, and that's in the ngtcp2 callback function where  ngtcp2 is asking use to handle the advertisement of the preferred address. We wrap the bits that ngtcp2 gives us (the ngtcp2_path and ngtcp2_preferred_addr pointers) into a `PreferredAddress` and pass that by const reference into the handler.

However, an argument can be made that use of the const reference here (and the Use) method goes against our c++ style guide, and that the PreferredAddress should be passed by pointer instead.  If that's the case, we can remove the `const` off `Use()` and the `mutable` off  `ngtcp2_path* dest`.

Reply:
Actually... I think I'll go ahead and make those changes rather than waiting for someone to complain ;-)